src/fsharp/CompileOps.fs

```diff
@@ -4889,7 +4903,83 @@ module private ScriptPreprocessClosure =
+        let appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)
+        let paketDepsFile,packageManagerTextLines =
+            let rec findDepsFile dir =
+                let fi = FileInfo(Path.Combine(dir,"paket.dependencies"))
+                if fi.Exists then
+                    fi,[ yield! Array.toList (File.ReadAllLines fi.FullName); yield! packageManagerTextLines]
+                elif fi.Directory.Parent <> null then
+                    findDepsFile fi.Directory.Parent.FullName
+                else
+                    let tempDir = Path.Combine(Path.Combine(Path.GetTempPath(),"fsharp"),"packages"+string(hash scriptName))
+                    if not (Directory.Exists tempDir) then
+                        Directory.CreateDirectory tempDir |> ignore
+                    let fi = FileInfo(Path.Combine(tempDir,"paket.dependencies"))
+                    File.WriteAllText(fi.FullName,"source https://nuget.org/api/v2" + Environment.NewLine)
+                    fi,[ yield "framework: net461"; yield "source https://nuget.org/api/v2"; yield! packageManagerTextLines]
+           
+            findDepsFile implicitIncludeDir
+
+        let workingDir = paketDepsFile.Directory.FullName
+
+        let paketExePathOpt = 
+            let rec loop dir = 
+                if (Directory.Exists dir) then 
+                    let paketDir = Path.Combine (dir, ".paket")
+                    if Directory.Exists paketDir then 
+                        let paketPath = Path.Combine (paketDir, PaketToolName)
+                        if File.Exists paketPath then Some paketPath else
+                        let paketPath = Path.Combine (dir, PaketToolName)
+                        if File.Exists paketPath then Some paketPath else
+                        None
+                    else
+                        None
+                else
+                    None
+
+            match loop implicitIncludeDir with 
+            | Some r -> Some r
+            | None -> 
+                match loop workingDir with 
+                | Some r -> Some r
+                | None -> 
+                     let paketPath = Path.Combine (appData, ".paket", PaketToolName)
+                     if File.Exists paketPath then Some paketPath else None
+        
+        match paketExePathOpt with 
+        | None -> 
+            errorR(Error(FSComp.SR.packageManagerNotFound(implicitIncludeDir, appData),m))
+            None
+
+        | Some paketExePath ->
+
+            match packageManagerExecTable.TryGetValue((paketDepsFile.FullName, paketExePath)) with 
+            | (true, (t, loadScript)) when File.Exists paketDepsFile.FullName && File.Exists loadScript && (t = packageManagerTextLines) -> 
+                printfn "skipping running package resolution... already done that"
+                Some loadScript
+
+            | _ ->
+                File.WriteAllLines(paketDepsFile.FullName, packageManagerTextLines)
```

Comment:
This seems to overwrite the existing ``paket.dependencies``

Or was the idea that simply adding ``#r "paket"`` would get you the packages listed there?